packaging/macos/tests/test_packaging_assets.sh

```
@@ -0,0 +1,184 @@
+  bundle="$(cd "${bundle}" && pwd -P)"
+  mkdir -p "${bundle}/lib"
+  cp "${PKG_DIR}/install.sh"           "${bundle}/install.sh"
+  cp "${PKG_DIR}/lib/installer_lib.sh" "${bundle}/lib/installer_lib.sh"
+  printf '<?xml version="1.0"?><plist/>' > "${bundle}/com.defenseclaw.gateway.plist"
+  chmod 0755 "${bundle}/install.sh"
+  if [[ "${with_binary}" == "true" ]]; then
+    printf '#!/bin/sh\nexit 0\n' > "${bundle}/defenseclaw-gateway"
+    chmod 0755 "${bundle}/defenseclaw-gateway"
+  fi
+  # Bypass the root check so tests can reach the resolution block.
+  sed -i.bak 's/\[\[ \$EUID -eq 0 \]\] || die/[[ 1 -eq 0 ]] || : /' "${bundle}/install.sh"
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)